### PR TITLE
Rename DockerSpawner.whitelist to allowed

### PR DESCRIPTION
**Pull request by GeorgianaElena**
cf https://github.com/jupyterhub/team-compass/issues/299
https://github.com/jupyterhub/jupyterhub/pull/3090

- [x] DockerSpawner.image_whitelist -> DockerSpawner.allowed_images
- [x] DockerSpawner.check_image_whitelist -> DockerSpawner.check_allowed
- [x] deprecate traits, config
- [x] test deprecated names still work
- [x] test deprecated config
- [x] update docs (no occurrences in docs)

### CODE REVIEW

File: dockerspawner/dockerspawner.py [hunk 1]
```diff
@@ -54,8 +54,72 @@ def validate(self, obj, value):
 class DockerSpawner(Spawner):
     """A Spawner for JupyterHub that runs each user's server in a separate docker container"""
 
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._init_deprecated_methods()
+
+    def _init_deprecated_methods(self):
```
Comment (minrk) [inline dockerspawner/dockerspawner.py:61]: Unlike the Authenticators, where subclassing and overriding methods is a standard approach, I don't think subclassing and overriding these methods is something we need to keep on DockerSpawner.

File: dockerspawner/dockerspawner.py [hunk 2]
```diff
@@ -54,8 +54,72 @@ def validate(self, obj, value):
...
+                def _new_calls_old(old_name, *args, **kwargs):
+                    return getattr(self, old_name)(*args, **kwargs)
+
+                setattr(self, new_name, partial(_new_calls_old, old_name))
+                break
+
     _executor = None
 
+    _deprecated_aliases = {
+        "container_ip": ("host_ip", "0.9.*"),
```
Comment (minrk) [inline dockerspawner/dockerspawner.py:93]: Great job consolidating these!

File: dockerspawner/dockerspawner.py (hunk 1 above)
Comment (GeorgianaElena) [reply dockerspawner/dockerspawner.py]: It makes sense. Thanks for explaining this, Min!

**Comment by minrk** [comment]
Excellent, thank you!

**Pull request merged**